managed/resource_cluster.go

```diff
@@ -844,8 +1064,26 @@ func (r resourceCluster) Read(ctx context.Context, req tfsdk.ReadResourceRequest
 	if state.BackupSchedules != nil && len(state.BackupSchedules) > 0 {
 		backUpSchedules = append(backUpSchedules, state.BackupSchedules[0])
 	}
+
 	cluster, readOK, message := resourceClusterRead(ctx, state.AccountID.Value, state.ProjectID.Value, state.ClusterID.Value, backUpSchedules, regions, allowListProvided, allowListIDs, false, r.p.client)
 
+	// Fetch the cmkSpec information from State (to get unmasked creds)
+	var cmkSpec CMKSpec
+	req.State.GetAttribute(ctx, path.Root("cmk_spec"), &cmkSpec)
+
+	if cluster.CMKSpec != nil {
+		// Unmask the creds to store in the State file
+		providerType := cluster.CMKSpec.ProviderType.Value
+		switch providerType {
+		case "AWS":
+			cluster.CMKSpec.AWSCMKSpec.SecretKey.Value = cmkSpec.AWSCMKSpec.SecretKey.Value
+			cluster.CMKSpec.AWSCMKSpec.AccessKey.Value = cmkSpec.AWSCMKSpec.AccessKey.Value
+		case "GCP":
+			cluster.CMKSpec.GCPCMKSpec.GcpServiceAccount.ClientId.Value = cmkSpec.GCPCMKSpec.GcpServiceAccount.ClientId.Value
+			cluster.CMKSpec.GCPCMKSpec.GcpServiceAccount.PrivateKey.Value = cmkSpec.GCPCMKSpec.GcpServiceAccount.PrivateKey.Value
+		}
+	}
```

Comment:
Did you try applying the same template multiple times? I believe you are not populating `is_enabled`, so it might break...

Reply:
Yes, it works when I apply the same template multiple times. 
`is_enabled` is populated here inside the `resourceClusterRead` function. 

https://github.com/yugabyte/terraform-provider-ybm/blob/86be15a353ea7f62feee3944ed4fa4a1492d0147/managed/resource_cluster.go#L1163

The code being referred here assigns the credentials field (secret key/access key, client id/private key) to their original value (since the GET API returns a masked value).